spi/spi_master_slave/spi_slave/spi_slave.c

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright (c) 2021 Michael Stoops. All rights reserved.
+ * Portions copyright (c) 2021 Raspberry Pi (Trading) Ltd.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+// Example of an SPI bus slave using the PL022 SPI interface
+
+#include <stdio.h>
+#include <string.h>
+#include "pico/stdlib.h"
+#include "pico/binary_info.h"
+#include "hardware/spi.h"
+
+#define BUF_LEN         0x100
+
+void printbuf(uint8_t buf[], size_t len) {
+    int i;
+    for (i = 0; i < len; ++i) {
+        if (i % 16 == 15)
+            printf("%02x\n", buf[i]);
+        else
+            printf("%02x ", buf[i]);
+    }
+
+    // append trailing newline if there isn't one
+    if (i % 16) {
+        putchar('\n');
+    }
+}
+
+
+int main() {
+    // Enable UART so we can print
+    stdio_init_all();
+#if !defined(spi_default) || !defined(PICO_DEFAULT_SPI_SCK_PIN) || !defined(PICO_DEFAULT_SPI_TX_PIN) || !defined(PICO_DEFAULT_SPI_RX_PIN) || !defined(PICO_DEFAULT_SPI_CSN_PIN)
+#warning spi/spi_slave example requires a board with SPI pins
+    puts("Default SPI pins were not defined");
+#else
+
+    printf("SPI slave example\n");
+
+    // Enable SPI 0 at 1 KHz and connect to GPIOs
+    spi_init(spi_default, 1000);
```

Comment:
Note this frequency is unreachable from a system clock of 125 MHz, and in fact your trace shows an SCK of ~2 kHz, with the SPI's divisors maxed out at 254 * 256. This would trip the assertions in spi.c if they were enabled.

1 MHz is fine and fairly conservative

Reply:
I guess this comment should really have been tagged on `spi_master.c` :)

Reply:
Thanks for the info, I'll change it.

My intention was to keep the process from happening too quickly. I guess I'll bump up the baudrate and just add some `sleep`.